src/samples.jl

```diff
@@ -576,7 +576,13 @@ function Arrow.ArrowTypes.JuliaType(::Val{SAMPLES_ARROW_NAME}, ::Type{<:SamplesA
 end
 
 function Arrow.ArrowTypes.fromarrow(::Type{<:Samples}, arrow_data, arrow_info, arrow_encoded)
-    info = Arrow.ArrowTypes.fromarrow(SamplesInfoV2, arrow_info...)
+    info = Arrow.ArrowTypes.fromarrow(SamplesInfoV2, arrow_info)
     data = reshape(arrow_data, (channel_count(info), :))
     return Samples(data, info, arrow_encoded)
 end
+
+# XXX in Arrow 2.7, the extra indirection with `fromarrowstruct` leads to the named tuple
+# being splatted as positional arguments and not keyword args unless we add this method.
```

Comment:
I might be missing something obvious here, but looking at https://github.com/apache/arrow-julia/pull/493/files, I don't see how it was ever splatting it as keyword arguments 🤔. Is that right, or was it something else?